runtime/calamari/src/lib.rs

```diff
@@ -998,14 +1000,101 @@ impl_runtime_apis! {
+                }
+
+                fn worst_case_holding() -> MultiAssets {
+                    // A mix of fungible, non-fungible, and concrete assets.
+                    const HOLDING_FUNGIBLES: u32 = 100;
+                    const HOLDING_NON_FUNGIBLES: u32 = 100;
+                    let fungibles_amount: u128 = 100;
+                    let mut assets = (0..HOLDING_FUNGIBLES)
+                        .map(|i| {
+                            MultiAsset {
+                                id: Concrete(GeneralIndex(i as u128).into()),
+                                fun: Fungible(fungibles_amount * i as u128),
+                            }
+                        })
+                        .chain(core::iter::once(MultiAsset { id: Concrete(Here.into()), fun: Fungible(u128::MAX) }))
+                        .chain((0..HOLDING_NON_FUNGIBLES).map(|i| MultiAsset {
+                            id: Concrete(GeneralIndex(i as u128).into()),
+                            fun: NonFungible(asset_instance_from(i)),
+                        }))
+                        .collect::<Vec<_>>();
+
+                        assets.push(MultiAsset{
+                            id: Concrete(KmaLocation::get()),
+                            fun: Fungible(1_000_000 * KMA),
+                        });
+                        assets.into()
+                }
+            }
+
+            impl pallet_xcm_benchmarks::fungible::Config for Runtime {
+                type TransactAsset = Balances;
+
+                type CheckedAccount = CheckedAccount;
+                type TrustedTeleporter = TrustedTeleporter;
+                type TrustedReserve = TrustedReserve;
+
+                fn get_multi_asset() -> MultiAsset {
+                    MultiAsset {
+                        id: Concrete(KmaLocation::get()),
+                        fun: Fungible(1 * KMA),
+                    }
+                }
+            }
+
+            impl pallet_xcm_benchmarks::generic::Config for Runtime {
+                type Call = Call;
+
+                fn worst_case_response() -> (u64, Response) {
+                    (0u64, Response::Version(Default::default()))
+                }
+
+                fn transact_origin() -> Result<MultiLocation, BenchmarkError> {
+                    Ok(KsmLocation::get())
+                }
+
+                fn subscribe_origin() -> Result<MultiLocation, BenchmarkError> {
+                    Ok(KsmLocation::get())
+                }
+
+                fn claimable_asset() -> Result<(MultiLocation, MultiLocation, MultiAssets), BenchmarkError> {
+                    let origin = KmaLocation::get();
+                    let assets: MultiAssets = (Concrete(KmaLocation::get()), 1_000 * KMA).into();
+                    let ticket = MultiLocation { parents: 0, interior: Here };
+                    Ok((origin, ticket, assets))
+                }
+            }
+
             let whitelist: Vec<TrackedStorageKey> = vec![
```

Comment:
Is there any xcm key we should whitelist?

Reply:
I don't think so.
On Kusama they whitelist storage read to `ActiveConfig` https://github.com/paritytech/polkadot/pull/4442/files#diff-fc698698c7826b51219876977c4813bc7d173e8ecc1cdb963ca422ea3d300427R112
But this doesn't show up in my weights results, and anyway it whitelisting doesn't change my results noticeably.